app/tests/reader_studies_tests/test_models.py

```diff
@@ -570,3 +650,116 @@ def test_clean_question_widget(answer_type, widget, interface, error):
 
     with error:
         q._clean_widget()
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "widget, options, error",
+    (
+        (
+            QuestionWidgetKindChoices.NUMBER_INPUT,
+            {},
+            nullcontext(),
+        ),
+        (
+            QuestionWidgetKindChoices.NUMBER_INPUT,
+            {"answer_min_value": 1},
+            nullcontext(),
+        ),
+        (
+            QuestionWidgetKindChoices.NUMBER_INPUT,
+            {"answer_max_value": 5},
+            nullcontext(),
+        ),
+        (
+            QuestionWidgetKindChoices.NUMBER_INPUT,
+            {"answer_min_value": -1},
+            nullcontext(),
+        ),
+        (
+            QuestionWidgetKindChoices.NUMBER_INPUT,
+            {"answer_min_value": 0},
+            nullcontext(),
+        ),
+        (
+            QuestionWidgetKindChoices.ACCEPT_REJECT,
+            {"answer_min_value": 1},
+            pytest.raises(ValidationError),
+        ),
+        (
+            "",
+            {"answer_min_value": 1},
+            pytest.raises(ValidationError),
+        ),
+        (
+            "",
+            {"answer_min_value": 0},
+            pytest.raises(ValidationError),
+        ),
+        (
+            QuestionWidgetKindChoices.NUMBER_INPUT,
+            {"answer_step_size": 1},
+            pytest.raises(ValidationError),
+        ),
+        (
+            QuestionWidgetKindChoices.NUMBER_INPUT,
+            {"answer_step_size": 0.5, "answer_min_value": 0},
+            nullcontext(),
+        ),
+        (
+            QuestionWidgetKindChoices.NUMBER_INPUT,
+            {"answer_step_size": -1, "answer_min_value": 0},
+            pytest.raises(ValidationError),
+        ),
+        (
+            QuestionWidgetKindChoices.NUMBER_INPUT,
+            {"answer_step_size": 0, "answer_min_value": 0},
+            pytest.raises(ValidationError),
+        ),
+        (
+            QuestionWidgetKindChoices.NUMBER_INPUT,
+            {
+                "answer_step_size": 0.3,
+                "answer_min_value": 0,
+                "answer_max_value": 4,
+            },
+            pytest.raises(ValidationError),
+        ),
+        (
+            QuestionWidgetKindChoices.NUMBER_INPUT,
+            {
+                "answer_step_size": 0.5,
+                "answer_min_value": 0,
+                "answer_max_value": 4,
+            },
+            nullcontext(),
+        ),
+        (
+            QuestionWidgetKindChoices.NUMBER_INPUT,
+            {
+                "answer_min_value": 4,
+                "answer_max_value": 0,
+            },
+            pytest.raises(ValidationError),
+        ),
+        (
+            QuestionWidgetKindChoices.NUMBER_INPUT,
+            {
+                "answer_min_value": 0,
+                "answer_max_value": 0,
+            },
+            pytest.raises(ValidationError),
+        ),
+    ),
+)
+def test_clean_widget_options(widget, options, error):
```

Comment:
This test would be easier to debug if you also included part of the validation error you expect to find, at the moment Pytest only tells you that test n failed.

Reply:
I have also removed the test that checks the database constraints for the step_size (that it needs to be higher than 0.001). We don't usually test those types of constraints since we can trust Django to enforce them.